tools/psa/build_tfm.py

```diff
@@ -0,0 +1,143 @@
+
+dependencies = {
+    "trusted-firmware-m":
+    ['https://git.trustedfirmware.org/trusted-firmware-m.git',
+     'feature-twincpu'],
+    "mbedtls": ['https://github.com/ARMmbed/mbedtls.git',
+                'mbedtls-2.7.9'],
+    "mbed-crypto": ['https://github.com/ARMmbed/mbed-crypto.git',
+                    'mbedcrypto-1.1.0'],
+    "CMSIS_5": ['https://github.com/ARM-software/CMSIS_5.git', '5.5.0'],
+}
+
+def run_cmd_and_return_output(command):
+    """
+    Run the command in the sytem and return output.
+    Commands are passed as a list of tokens.
+    E.g. The command 'git remote -v' would be passed in as:
+     ['git', 'remote', '-v']
+
+    :param command: System command as a list of tokens
+    """
+    output = ''
+    logger.debug('[Exec] %s', ' '.join(command))
+    try:
+        fnull = open(os.devnull, 'w')
+        output = subprocess.check_output(command, stderr=fnull)
+    except subprocess.CalledProcessError as e:
+        logger.error("The command %s failed with return code: %s",
+                     (' '.join(command)), e.returncode)
+        sys.exit(1)
+    return output
+
+def detect_and_write_tfm_version(tfm_dir):
+    """
+    Identify the version of TF-M and write it to VERSION.txt
+    :param tfm_dir: The filesystem path where TF-M repo is cloned
+    """
+    cmd = ['git', '-C', tfm_dir, 'describe', '--tags',
+           '--abbrev=12', '--dirty', '--always']
+    tfm_version = run_cmd_and_return_output(cmd)
+    logger.info('TF-M version: %s', tfm_version)
+    if not isdir(VERSION_FILE_PATH):
+        os.makedirs(VERSION_FILE_PATH)
+    with open(join(VERSION_FILE_PATH, 'VERSION.txt'), 'w+') as f:
+        f.write(tfm_version)
+
+def check_repo_version(name, deps):
+    """
+    Compare the version of cloned and expected and exit if they dont match
+    :param name: Name of the git repository
+    :param deps: Dictionary containing dependency details
+    """
+    basedir = TF_M_BUILD_DIR
+    if name == 'trusted-firmware-m':
+        cmd = ['git', '-C', join(basedir, name),
+               'rev-parse', '--abbrev-ref', 'HEAD']
+    else:
+        cmd = ['git', '-C', join(basedir, name),
+               'describe', '--tags']
+    _out = run_cmd_and_return_output(cmd)
+    if _out.strip('\n') != deps.get(name)[1]:
+        logger.error('Conflict: cloned "%s" and expected "%s"',
+                     _out.strip('\n'), deps.get(name)[1])
+        logger.error('check and remove folder %s',
+                     join(basedir, name))
+        sys.exit(1)
+    else:
+        logger.info('%s: version check OK', name)
+
+def check_and_clone_repo(name, deps):
+    """
+    Test if the repositories are already cloned. If not clone them
+    :param name: Name of the git repository
+    :param deps: Dictionary containing dependency details
+    """
+    basedir = TF_M_BUILD_DIR
+    if not isdir(join(basedir, name)):
+        logger.info('Cloning %s repo', name)
+        cmd = ['git', '-C', basedir, 'clone', '-b',
```

Comment:
This (and other command calls to git) make assumptions that git is installed on the user's machine. None of this will work in the online build system either.

Reply:
Added is_git_installed() and is_git_lfs_installed() to check the availability of required tools.